storage/ndb/src/kernel/blocks/dbtup/AggInterpreter.cpp

```diff
@@ -0,0 +1,2138 @@
+  }
+
+  /*
+   * 4. Reset all aggregation results
+   */
+  if (n_agg_results_) {
+#ifdef MOZ_AGG_MALLOC
+    assert(n_agg_results_ <= MAX_AGG_N_RESULTS);
+    agg_results_ = agg_results_buf_;
+#else
+    agg_results_ = new AggResItem[n_agg_results_];
+#endif // MOZ_AGG_MALLOC
+    uint32_t i = 0;
+    while (i < n_agg_results_) {
+      agg_results_[i].type = NDB_TYPE_UNDEFINED;
+      agg_results_[i].value.val_int64 = 0;
+      agg_results_[i].is_unsigned = false;
+      agg_results_[i].is_null = true;
+      i++;
+    }
+  }
+
+  inited_ = true;
+  agg_prog_start_pos_ = cur_pos_;
+  memset(registers_, 0, sizeof(registers_));
+
+  return true;
+}
+
+static bool TypeSupported(DataType type) {
+  switch (type) {
+    case NDB_TYPE_TINYINT:
+    case NDB_TYPE_SMALLINT:
+    case NDB_TYPE_MEDIUMINT:
+    case NDB_TYPE_INT:
+    case NDB_TYPE_BIGINT:
+
+    case NDB_TYPE_TINYUNSIGNED:
+    case NDB_TYPE_SMALLUNSIGNED:
+    case NDB_TYPE_MEDIUMUNSIGNED:
+    case NDB_TYPE_UNSIGNED:
+    case NDB_TYPE_BIGUNSIGNED:
+
+    case NDB_TYPE_FLOAT:
+    case NDB_TYPE_DOUBLE:
+
+    case NDB_TYPE_DECIMAL:
+    case NDB_TYPE_DECIMALUNSIGNED:
+      return true;
+    default:
+      return false;
+  }
+  return false;
+}
+
+static bool IsUnsigned(DataType type) {
+  switch (type) {
+    case NDB_TYPE_TINYUNSIGNED:
+    case NDB_TYPE_SMALLUNSIGNED:
+    case NDB_TYPE_MEDIUMUNSIGNED:
+    case NDB_TYPE_UNSIGNED:
+    case NDB_TYPE_BIGUNSIGNED:
+    case NDB_TYPE_DECIMALUNSIGNED:
+      return true;
+    default:
+      return false;
+  }
+  return false;
+}
+
+static DataType AlignedType(DataType type) {
+  switch (type) {
+    case NDB_TYPE_TINYINT:
+    case NDB_TYPE_SMALLINT:
+    case NDB_TYPE_MEDIUMINT:
+    case NDB_TYPE_INT:
+    case NDB_TYPE_BIGINT:
+
+    case NDB_TYPE_TINYUNSIGNED:
+    case NDB_TYPE_SMALLUNSIGNED:
+    case NDB_TYPE_MEDIUMUNSIGNED:
+    case NDB_TYPE_UNSIGNED:
+    case NDB_TYPE_BIGUNSIGNED:
+
+      return NDB_TYPE_BIGINT;
+    case NDB_TYPE_FLOAT:
+    case NDB_TYPE_DOUBLE:
+      return NDB_TYPE_DOUBLE;
+    /*
+     * TODO (Zhao)
+     * Moz
+     * Temporary solultion
+     * Currently regard Decimal as a undefined,
+     * then decide it as BIGINT/BIGUNSIGNED/DOUBLE in LoadColumn
+     * dynamically.
+     */
+    case NDB_TYPE_DECIMAL:
+    case NDB_TYPE_DECIMALUNSIGNED:
+      return NDB_TYPE_UNDEFINED;
+    default:
+      assert(0);
+  }
+  return NDB_TYPE_UNDEFINED;
+}
+
+static bool TestIfSumOverflowsUint64(uint64_t arg1, uint64_t arg2) {
+  return ULLONG_MAX - arg1 < arg2;
+}
+
+static void SetRegisterNull(Register* reg) {
+  reg->is_null = true;
+  reg->value.val_int64 = 0;
+  reg->is_unsigned = false;
+}
+
+static void ResetRegister(Register* reg) {
+  reg->type = NDB_TYPE_UNDEFINED;
+  SetRegisterNull(reg);
+}
+
+static int32_t RegPlusReg(const Register& a, const Register& b, Register* res) {
+  assert(a.type != NDB_TYPE_UNDEFINED && b.type != NDB_TYPE_UNDEFINED);
+
+  DataType res_type = NDB_TYPE_UNDEFINED;
```

Comment:
No need to assign it UNDEFINED, it will be assigned in both if and else.

Reply:
I prefer to keep it. Just follow Google code guide: ”initialize variables in the declaration“